greenpithumb/greenpithumb.py

```diff
@@ -114,7 +115,10 @@ def main(args):
             for current_poller in pollers:
                 current_poller.start_polling_async()
             while True:
-                record_processor.process_next_record()
+                record_processor.try_process_next_record()
+                time.sleep(0.5)
```

Comment:
Otherwise it spikes the CPU because the program has nothing to do except keep calling `try_process_next_record` nonstop. With `sleep` the OS knows the program has nothing to do, so it doesn't use the CPU in that thread.